### ISSUE TITLE
Questions about `sft_train.py`: Learning Rate, and Slow Learning of Layer 0 exit probabilities

### ISSUE THREAD

**Issue by karthikviswanathn** [MeridianResearch/externalization#12]
Hi @puria-radmard, @lizardp1 and @marichkakorolyuk,

I have a few questions regarding the `sft_train.py` script:

1. **Learning Rate Guidelines:**
   - Are there any recommended guidelines for setting the learning rate? For example, is `lr = 1e-5` sufficient, or could it be too low for the linear weights?

2. **Slow Learning of Layer 0 Probabilities**
   - P(exit) $\sim$ 0 for layer 0, however it takes a long time for the early exit decision heads to learn this. For example, in [this run training on a SINGLE prompt](https://wandb.ai/vkarthik095-university-of-amsterdam/early-exit/runs/s388kxvy?nw=nwuservkarthik095), the layer 0 exit probabilities start at 0.5 and gradually decrease as seen from the `layer_mean_exit_probabilities_student` chart. The gradients for the layer 0 early exit weights were also much smaller compared to the other layers as seen below.
   - P(exit) = 0 seems to be a simple enough problem to learn, why does it take so many batches to learn this for layer 0?
   - Is a linear NN enough to estimate P(exit)? 

Here are some example gradient norms from the run:
> Layer 0: weight grad norm = 0.12
> Layer 5: weight grad norm = 14.1
> Layer 10: weight grad norm = 18.20
> Layer 15: weight grad norm = 24.32
> Layer 20: weight grad norm = 25.29
> Layer 25: weight grad norm = 24.11

3. **Logging hidden states in sft_teacher**
```
self.exit_state.log_residual_stream(
                        hidden_states = hidden_states,
                        layer_idx = self.layer_idx,
                        exitable_layer_idx = self.exitable_layer_idx,
                    )
layer_output = super().forward(hidden_states = hidden_states, **kwargs) 
```
Just wanted to double-check if we want to log `layer_output` or`hidden_states`.

Feel free to let me know if I should elaborate on any of these points!
